### ISSUE TITLE
`unbound-method` lint message has a problematic newline

### ISSUE THREAD

**Issue by zsiegel92** [typescript-eslint/typescript-eslint#11459]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://github.com/current-ai-llc/current-ai/pull/2162

### Repro Code

```TypeScript
Count number of each type of error/warning:

<command to run your linter>  \
  | sed -E 's/^.*(Error|Warning):.*(@typescript-eslint\/[^[:space:]]*).*/\1 \2/' \
  | sort \
  | uniq -c \
  | sort -nr
```

### ESLint Config

```javascript
(any eslint config)
```

### tsconfig

```jsonc
(any tsconfig)
```

### Expected Result

I would expect to see a line like

```
36 Error @typescript-eslint/unbound-method
```

### Actual Result

```
...
  59 Error @typescript-eslint/no-unsafe-return
  36 If your function does not access `this`, you can annotate it with `this: void`, or consider using an arrow function instead.  @typescript-eslint/unbound-method
  13 Error @typescript-eslint/no-redundant-type-constituents
...
```

### Additional Info

[This PR](https://github.com/typescript-eslint/typescript-eslint/pull/11458) fixes this.

**Comment by bradzacher** [comment]
this newline is intentional.
if you want to programmatically parse the ESLint output -- don't use the human readable text output, use ESLint's programmatically parseable JSON output `--format json`

**Comment by zsiegel92** [comment]
@bradzacher thanks for taking a look. Yes I had already started using json but thought it was a bug nonetheless.

Just curious - what's the argument for the newline in this message when most (all?) other typescript-eslint error messages are on a single line (and amenable to parsing in a unix-y way)

**Comment by bradzacher** [comment]
It improves readability. There's no restriction in eslint itself to only use a single line. So we can provide a better DevX with it.
